reference-lib/index.ts

```diff
@@ -41,37 +47,63 @@ for (const modules of reference.modules) {
  * Returns all the nginx directives
  *
  *  @param: format: format of the return type HTML or markdown
- *
  *  @return: an array of Directives
  */
-export function getDirectives(format=Format.HTML): Directive[] {
-    const directives = reference.modules.flatMap((m) =>
-      m.directives.map((d) => ({...d, module: m.name})))
-    .map ((d) => ({
-        name: d.name,
-        module: d.module,
-        description: format === Format.HTML ? d.description_html : d.description_md,
-        syntax: format === Format.HTML ? d.syntax_html : d.syntax_md,
-        contexts: d.contexts,
-        isBlock: d.isBlock,
-        default: d.default
-    } as Directive))
-    return directives
+export function getDirectives(format = Format.HTML): Directive[] {
```

Comment:
I kept `Variable` and `Directive` separate, but based on usage I'm wondering if we want to return variables from here too, and add a `isVariable` or something so callers can make decisions if they want to.

Thoughts?

Reply:
as some Slack and thinking further, I like this split better than pretending variables are directives. They can be easily combined by consumers if needed.

Going to update the autocomplete example to demo that